extensions/wikia/WikiFactory/Hubs/WikiFactoryHub.php

```diff
@@ -186,6 +185,16 @@ public function getVerticalId( $city_id ) {
 		return $id->city_vertical;
 	}
 
+	public function getVerticalNameMessage( $verticalId ) {
+		$message = false;
+		$verticals = $this->getAllVerticals();
+		if ( isset( $verticals[$verticalId] ) ) {
+			$message = wfMessage( 'vertical-' . $verticals[$verticalId]['short'] );
```

Comment:
Concatenating message key makes grepping it in codebase impossible. Please note Grunny's email on Messages